components/layout_2020/replaced.rs

```diff
@@ -554,119 +554,34 @@ impl ReplacedContent {
-                    if transferred_block_size <= max_block_size {
-                        // Row 6.
-                        LogicalVec2 {
-                            inline: max_inline_size,
-                            block: transferred_block_size.max(min_box_size.block),
-                        }
-                    } else {
-                        // Row 7.
-                        LogicalVec2 {
-                            inline: ratio
-                                .compute_dependent_size(Direction::Inline, max_block_size)
-                                .max(min_box_size.inline),
-                            block: max_block_size,
-                        }
-                    }
-                },
-                // Rows 8-9.
-                (Violation::Below(min_inline_size), Violation::Below(min_block_size)) => {
-                    let transferred_inline_size =
-                        ratio.compute_dependent_size(Direction::Inline, min_block_size);
-                    if min_inline_size <= transferred_inline_size {
-                        // Row 8.
-                        LogicalVec2 {
-                            inline: transferred_inline_size.clamp_below_max(max_box_size.inline),
-                            block: min_block_size,
-                        }
-                    } else {
-                        // Row 9.
-                        LogicalVec2 {
-                            inline: min_inline_size,
-                            block: ratio
-                                .compute_dependent_size(Direction::Block, min_inline_size)
-                                .clamp_below_max(max_box_size.block),
-                        }
-                    }
-                },
-                // Row 10.
-                (Violation::Below(min_inline_size), Violation::Above(max_block_size)) => {
-                    LogicalVec2 {
-                        inline: min_inline_size,
-                        block: max_block_size,
-                    }
-                },
-                // Row 11.
-                (Violation::Above(max_inline_size), Violation::Below(min_block_size)) => {
-                    LogicalVec2 {
-                        inline: max_inline_size,
-                        block: min_block_size,
-                    }
-                },
-            };
+            let tentative_size = get_tentative_size(box_size);
+            let max_box_size = max_box_size.map(|max_size| max_size.unwrap_or(MAX_AU));
+            // This is a simplification of the CSS2 algorithm in a way that properly handles `aspect-ratio`.
+            // We transfer min and max constraints from the other axis, and apply them in addition to
+            // non-transferred min and max constraints. In case of conflict,
+            //  - Non-transferred constraints take precedence over transferred ones.
+            //  - Min constraints take precedence over max ones from the same axis.
```

Comment:
Do you mind dropping a link to the relevant spec issue or comment here?

Reply:
Linked to https://drafts.csswg.org/css-sizing-4/#aspect-ratio-size-transfers

Reply:
Thanks! Do you think a link to <https://github.com/w3c/csswg-drafts/issues/6071#issuecomment-2243986313> makes sense as well?

Reply:
Sure, done